src/policy/CMSRucioPolicy/algorithms/auto_approve.py

```diff
@@ -81,18 +97,24 @@ def global_approval(did, rule_attributes, session):
     if rule_attributes['lifetime'] > RULE_LIFETIME_THRESHOLD:
         return False
 
+    size_of_rule = sum([file['bytes'] for file in list_files(did['scope'], did['name'], session=session)])
+
+    # Limit single RSE rules to 50 TB
+    # This does not mean that the total locks size at a RSE will be limited to 50 TB
+    # as other rules that are spread over multiple RSEs may claim the same space
+    # This is just a simple check to avoid a single RSE rules from being too large
+    rse_expression = rule_attributes['rse_expression']
+    rses = parse_expression(rse_expression, filter_={'availability_write': True}, session=session)
+    if len(rses) == 1:
+        this_rse_autoapprove_rules = list_rules(filters={'account': account, 'activity': AUTO_APPROVE_ACTIVITY, 'rse_expression': rse_expression}, session=session)
+        this_rse_autoapprove_usage = _get_rules_size(this_rse_autoapprove_rules)
+        if this_rse_autoapprove_usage + size_of_rule > SINGLE_RSE_RULE_SIZE_THRESHOLD:
+            return False
+
     # Check global usage of the account under this activity
-    data_brokering_usage = 0
-    data_brokering_rules = list_rules(filters={'account': account, 'activity': 'Data Brokering'}, session=session)
-    for db_rule in data_brokering_rules:
-        scope = db_rule['scope']
-        name = db_rule['name']
-        rule_files = list_files(scope, name, session=session)
-        rule_size = sum([file['bytes'] for file in rule_files])
-        data_brokering_usage += rule_size
-
-    this_rule_size = sum([file['bytes'] for file in list_files(did['scope'], did['name'], session=session)])
-    if data_brokering_usage + this_rule_size > GLOBAL_USAGE_THRESHOLD:
+    all_auto_approve_rules_by_account = list_rules(filters={'account': account, 'activity': AUTO_APPROVE_ACTIVITY}, session=session)
+    global_auto_approve_usage_by_account = _get_rules_size(all_auto_approve_rules_by_account)
```

Comment:
This looks like a per-account sum which should be OK. If you are looking to sum over all accounts, I think that's a little much. And you cannot write bare SQL queries. You'll need to get something added to the right Rucio model file to do what you need. 

That should be deferred until it can be done right. 

If you are unhappy with this PR, descope it. But we need to get it merged soon so we are not running production code off of your PR. We discussed this in chat.